### PR TITLE
Improves many detections, adds detection for various apps and adds detection for various bots

### PR DESCRIPTION
**Pull request by liviuconcioiu**

### CODE REVIEW

File: regexes/client/mobile_apps.yml [hunk 1]
```diff
@@ -1079,3 +1089,73 @@
...
+
+# Bank Millenium (https://apps.apple.com/us/app/bank-millennium/id412646128)
+- regex: '(Millennium/|Millennium%20Corp)'
+  name: 'Bank Millenium'
+  version: ''
+
+# MTrader (https://apps.apple.com/us/app/mtrader/id436829408)
+- regex: 'MBolsa'
+  name: 'MTrader'
+  version: ''
```
Comment (sanchezzzhak) [inline regexes/client/mobile_apps.yml:1121]: https://apps.apple.com/us/app/mbolsa/id1533621499
https://play.google.com/store/apps/details?id=co.mbolsa.app

name MBolsa

File: Tests/fixtures/console.yml [hunk 2]
```diff
@@ -379,7 +379,10 @@
   browser_family: Chrome
 -
   user_agent: Mozilla/5.0 (PlayStation; PlayStation 5/2.26) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/13.0 Safari/605.1.15
-  os: [ ]
+  os:
+    name: PlayStation
+    version: "5"
+    platform: ""
```
Comment (sanchezzzhak) [inline Tests/fixtures/console.yml:385]: It seems that each gen of consoles has its own versions software
https://www.reddit.com/r/playstation/wiki/guides/system-software/ps5

https://www.reddit.com/r/playstation/wiki/guides/system-software/ps4
https://en.wikipedia.org/wiki/PlayStation_4_system_software

here we need to think about how to specify the version correctly

`Mozilla/5.0 (PlayStation; PlayStation 5/2.26) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/13.0 Safari/605.1.15`
versions: 5/2.26 or 5.2.26 for PS5

`Mozilla/5.0 (PlayStation; PlayStation 4/9.03) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.0 Safari/605.1.15`
versions: 4/9.03 or 4.9.03 for PS4
`Mozilla/5.0 (PlayStation 4 3.11) AppleWebKit/537.73 (KHTML, like Gecko)`
versions: 4/3.11 or 4.3.11  for PS4

@sgiehl What version format is acceptable to us?

File: regexes/oss.yml [hunk 3]
```diff
@@ -335,12 +333,14 @@
   versions:
     - regex: 'WEBOS(\d+[\.\d]+)'
```
Comment (sanchezzzhak) [inline regexes/oss.yml:334]: added new version Web0S 6 (optional)
Mozilla/5.0 (Web0S; Linux/SmartTV) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.79 Safari/537.36 WebAppManager

File: Tests/fixtures/tv-1.yml [hunk 4]
```diff
@@ -4091,3 +4091,1000 @@
     model: X4S
   os_family: Android
   browser_family: Chrome
+-
+  user_agent: Opera/9.80 (Linux mips; Opera TV/39) Presto/2.11.355 Version/12.11
+  os:
+    name: GNU/Linux
+    version: ""
+    platform: MIPS
```
Comment (sanchezzzhak) [inline Tests/fixtures/tv-1.yml:4099]: (optional)
here we can define OS - Opera TV
https://blogs.opera.com/news/2016/01/ces-2016-meet-opera-tv/

regex: `Linux (?:[:^;]+); Opera TV(?: Store)?/`
os_family GNU/Linux

Opera/9.80 (Linux mips; Opera TV/4) Presto/2.11.355 Version/12.11
Opera/9.80 (Linux armv7l; Opera TV Store/5581) Presto/2.12.362 Version/12.11

File: Tests/Parser/Client/fixtures/browser.yml [hunk 5]
```diff
@@ -4229,3 +4229,102 @@
...
+    type: browser
+    name: Reqwireless WebViewer
+    version: 2.0.0.0
+    engine: ""
+    engine_version: ""
+    family:
+-
+  user_agent: Mozilla/5.0 (X11; Linux x86_64; rv:21.0) Gecko/20100101 SlimerJS/0.7
+  client:
+    type: browser
```
Comment (sanchezzzhak) [inline Tests/Parser/Client/fixtures/browser.yml:4298]: client type: lib

https://www.npmjs.com/package/slimerjs

File: DeviceDetector.php [hunk 6]
```diff
@@ -948,7 +948,7 @@ protected function parseDevice(): void
         /**
          * All devices running Opera TV Store are assumed to be a tv
          */
-        if ($this->matchUserAgent('Opera TV Store')) {
+        if ($this->matchUserAgent('Opera TV Store') || $this->matchUserAgent(' OMI/')) {
```
Comment (sanchezzzhak) [inline DeviceDetector.php:951]: combine two conditions into one
```
$this->matchUserAgent('Opera TV Store| OMI/')
```

File: Tests/fixtures/tv-1.yml [hunk 7]
```diff
@@ -4092,23 +4092,1200 @@
...
+  client:
+    type: browser
+    name: Opera
+    version: "12.51"
+    engine: Presto
+    engine_version: 2.12.407
+  device:
+    type: tv
+    brand: Polaroid
+    model: Si2157LG32
```
Comment (sanchezzzhak) [inline Tests/fixtures/tv-1.yml:4199]: model: ""

`Si2157LG32`, `Mxl661L32`, `` is a panel/mainboard  models that is installed on several brands
JVC, LG, MEDION, Polaroid, HITACHI, LUXOR etc

File: Tests/fixtures/tv-1.yml [hunk 8]
```diff
@@ -4092,23 +4092,1200 @@
...
+  client:
+    type: browser
+    name: Opera
+    version: "12.51"
+    engine: Presto
+    engine_version: 2.12.407
+  device:
+    type: tv
+    brand: Toshiba
+    model: Si2156LG32
```
Comment (sanchezzzhak) [inline Tests/fixtures/tv-1.yml:4181]: model: ""

File: Tests/fixtures/tv-1.yml [hunk 9]
```diff
@@ -4092,23 +4092,1200 @@
...
+  client:
+    type: browser
+    name: Opera Devices
+    version: 4.2.12.31
+    engine: Blink
+    engine_version: ""
+  device:
+    type: tv
+    brand: ""
+    model: Si2158LG32
```
Comment (sanchezzzhak) [inline Tests/fixtures/tv-1.yml:5268]: model: ""

File: Tests/fixtures/tv-1.yml [hunk 10]
```diff
@@ -4092,23 +4092,1200 @@
...
+  client:
+    type: browser
+    name: Opera Devices
+    version: 4.6.1.13
+    engine: Blink
+    engine_version: ""
+  device:
+    type: tv
+    brand: Philips
+    model: QM152E
```
Comment (sanchezzzhak) [inline Tests/fixtures/tv-1.yml:5232]: model: ""

image

QM152E - MainBoard or Version Software, used on Philips TVs for 2015-16

File: Tests/fixtures/tv-1.yml [hunk 11]
```diff
@@ -4092,23 +4092,1200 @@
...
+  client:
+    type: browser
+    name: Opera Devices
+    version: 4.21.0.273
+    engine: Blink
+    engine_version: ""
+  device:
+    type: tv
+    brand: Sony
+    model: BRAVIA-VH1
```
Comment (sanchezzzhak) [inline Tests/fixtures/tv-1.yml:5124]: Bravia VH1

**Review by sanchezzzhak — CHANGES_REQUESTED** [review]

File: Tests/fixtures/tv-1.yml [hunk 12]
```diff
@@ -4092,23 +4092,1200 @@
...
+  client:
+    type: browser
+    name: Opera
+    version: "12.51"
+    engine: Presto
+    engine_version: 2.12.407
+  device:
+    type: tv
+    brand: Crown
+    model: Mxl66140 V
```
Comment (sanchezzzhak) [inline Tests/fixtures/tv-1.yml:4926]: model: ""

**Review by sanchezzzhak — CHANGES_REQUESTED** [review]

**Review by sanchezzzhak — APPROVED** [review]

File: Tests/fixtures/console.yml (hunk 2 above)
Comment (sgiehl) [reply Tests/fixtures/console.yml:385]: As we don't use `/` in versions yet, maybe using a `.` instead would be better.

**Review by sgiehl — COMMENTED** [review]
Only had a rough look through the changes. Overall looks good to me.

File: Tests/fixtures/console.yml (hunk 2 above)
Comment (sanchezzzhak) [reply Tests/fixtures/console.yml:385]: @liviuconcioiu 
set full version

`Mozilla/5.0 (PlayStation; PlayStation 5/2.26) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/13.0 Safari/605.1.15`
``` yml
regex: 'PlayStation (\d)(?:[/ ](\d\.))?'
name: 'PlayStation'
vesion: '$1.$2'    // 5.2.26
```

**Review by sanchezzzhak — CHANGES_REQUESTED** [review]

**Review by sanchezzzhak — DISMISSED** [review]

File: regexes/oss.yml [hunk 13]
```diff
@@ -5,6 +5,24 @@
...
+  name: 'KreaTV'
+  version: '$1'
+
+##########
+# Opera TV (https://blogs.opera.com/news/2016/01/ces-2016-meet-opera-tv/)
+##########
+- regex: 'Linux (?:[^;]+); Opera TV(?: Store)?/|^Opera/\d+\.\d+ \(Linux mips'
+  name: 'Opera TV'
+  version: ''
+
```
Comment (sanchezzzhak) [inline regexes/oss.yml:25]: @sgiehl I want your opinion about this stuff?
I noticed that all TVs have UA
`^Opera/\d+\.\d+\(Linux mips` can we mark OS as Opera TV?

Comment (sanchezzzhak) [reply regexes/oss.yml:25]: find all tests by `\( linux mips`
image

result only findes tv's + opera

Comment (sgiehl) [reply regexes/oss.yml:25]: Then lets go with that for now. If at any time a useragent pops up that isn't detected correctly, we can always consider to change that again...

Comment (sanchezzzhak) [reply regexes/oss.yml:25]: merge?

**Review by sgiehl — APPROVED** [review]

**Pull request merged**